tests/integration/ha_tests/helpers.py

```diff
@@ -160,3 +145,43 @@ async def retrieve_entries(ops_test, app, db_name, collection_name, query_field)
 
     client.close()
     return cluster_entries
+
+
+async def find_unit(ops_test: OpsTest, leader: bool) -> ops.model.Unit:
+    """Helper function identifies the a unit, based on need for leader or non-leader."""
+    ret_unit = None
+    app = await cluster_name(ops_test)
+    for unit in ops_test.model.applications[app].units:
+        if await unit.is_leader_from_status() == leader:
+            ret_unit = unit
+
+    return ret_unit
+
+
+async def unit_ids(ops_test: OpsTest) -> List[int]:
+    """Provides a function for generating unit_ids in case a cluster is provided."""
+    provided_cluster = await cluster_name(ops_test)
+    if not provided_cluster:
+        return UNIT_IDS
+    unit_ids = [
+        unit.name.split("/")[1] for unit in ops_test.model.applications[provided_cluster].units
+    ]
+    return unit_ids
+
+
+async def cluster_name(ops_test: OpsTest) -> str:
```

Comment:
I found it a bit confusing calling this `cluster_name`, since what you are returning is the name the application was given on deployment for a given model. On contexts other than mongo, the cluster name can be a configuration of the workload itself.
How about calling this just `app_name`?

Reply:
Love that! Will do, this is a really good suggestion!!